TESTS/mbed_platform/minimal-printf/compliance/main.cpp

```diff
@@ -42,6 +42,94 @@
 
 using namespace utest::v1;
 
+#if defined(__NEWLIB_NANO)
+
+#define MAX_STRING_SIZE 100
+//Global buffer used by test cases to store string literal
+char expected_string[MAX_STRING_SIZE];
+
+typedef enum {
+    RADIX_10,//radix base for decimal
+    RADIX_16 //radix base for Hexadecimal
+} radix_base;
+
+// This function is similar to itoa but it can convert any data type with decimal and hexadecimal numerical value into a string
+static void convert_to_string(unsigned long long  value,  char str[],  radix_base radix = RADIX_10, bool sign = false)
+{
+    char        buffer[MAX_STRING_SIZE];
+    char       *dest = buffer + sizeof(buffer);
+
+    // If value is 0
+    if (value == 0) {
+        memcpy(str, "0", 2);
+        return;
+    }
+
+    if (sign) {
+        if ((long long) value < 0) {
+            // Make value postive
+            value = -value;
+        }
+    }
+    *--dest = '\0';
+    switch (radix) {
+        case RADIX_16:
+            while (value) {
+                * --dest = '0' + (value & 0xF);
+                if (*dest > '9') {
+                    *dest += 'A' - '9' - 1;
+                }
+                value >>= 4;
+            }
+            break;
+        case RADIX_10:
+            while (value) {
+                *--dest = '0' + (value % 10);
+                value /= 10;
+            }
+            break;
+
+        default:
+            MBED_ASSERT(0);
+            break;
+    }
+
+    if (sign) {
+        // Add the negative sign for negative numerical value
+        *--dest = '-';
+    }
+    // Copy the converted string from dest into str buffer of converted string length
+    memcpy(str, dest, buffer + sizeof(buffer) - dest);
+}
+
+//This function used to calculate the string length on format 'prefix + data + suffix'
+static int expected_string_length(char *prefix = nullptr, long long value, radix_base base, char *suffix = nullptr, bool sign = false)
```

Comment:
`is` missing from the sentence.
I think this function would beneft from having a function header with a description of the arguments. In particualr you should explain that `value` is converted into a string representing the integer with the given `base`.
You should also explain that you are using the global `expected_string` buffer to build the string.